usage-examples/javascript/mongosh/README.md

```diff
@@ -0,0 +1,157 @@
+
+1. Pull the MongoDB image from Docker Hub:
+
+   ```shell
+   docker pull mongo
+   ```
+2. Run the container:
+
+   ```shell
+   docker run --name mongodb-test -d -p 27017:27017 mongo
+   ```
+
+3. Verify the container is running:
+
+   ```shell
+   docker ps  
+   ```
+
+The output resembles:
+
+```shell
+CONTAINER ID   IMAGE   COMMAND                  CREATED        STATUS                  PORTS                        NAMES
+ef70cce38f26   mongo   "/usr/local/bin/runn…"   29 hours ago   Up 29 hours (healthy)   127.0.0.1:63201->27017/tcp   mongodb-test
+```
+
+You may note the actual port is different than `27017`, if something else is already running on
+`27017` on your machine. Note the port next to the IP address for running the tests.
+
+### Create a .env file
+
+Create a file named `.env` at the root of the `/mongosh` directory.
+Add the following values to your .env file, substituting the port where your local deployment
+is running:
+
+```
+CONNECTION_STRING="mongodb://localhost:63201"
+CONNECTION_PORT="63201"
+```
+
+### Install the dependencies
+
+This test suite requires you to have `Node.js` v20 or newer installed. If you
+do not yet have Node installed, refer to
+[the Node.js installation page](https://nodejs.org/en/download/package-manager)
+for details.
+
+From the root of the `/mongosh` directory, run the following command to install
+dependencies:
+
+```
+npm install
+```
+
+### Run the tests
+
+#### Run Tests from the IDE
+
+Normally, you could press the play button next to a test name to run a test
+in the IDE. Because this test suite relies on an Atlas connection string and
+environment value passed in from the environment, running tests in the IDE
+will fail unless you configure the IDE with the appropriate environment
+variables.
+
+In JetBrains IDEs, you can do the following:
+
+- Click the play button next to the test suite name
+- Select the `Modify Run Configuration` option
+- In the `Environment Variables` field, supply the appropriate environment variables
+    - Note: you do not need to use quotes around the connection string in this field
+      i.e. it should resemble:
+      CONNECTION_STRING=mongodb://localhost:63201
+
+#### Run All Tests from the command line
+
+From the `/mongosh` directory, run:
+
+```
+npm test
+```
+
+This invokes the following command from the `package.json` `test` key:
+
+```
+jest --run-in-band --detectOpenHandles
+```
+
+In the above command:
+
+- `jest` is the command to run the test suite
+- `--runInBand` is a flag that specifies only running one test at a time
+  to avoid collisions when creating/editing/dropping indexes. Otherwise, Jest
+  defaults to running tests in parallel.
+- `--detectOpenHandles` is a flag that tells Jest to track resource handles or async
+  operations that remain open after the tests are complete. These can cause the test suite
+  to hang, and this flag tells Jest to report info about these instances.
+
+#### Run Test Suites from the command line
```

Comment:
why aren't we using npm scripts here? would let us simplify these last sections to running the following from `/mongosh` root without requiring jest installed: 

```
npm test -- -t '<text string from the 'describe()' block you want to run>'
```
or 
```
npm test -- -t '<text string from the 'it()' block you want to run>'
```

...if we do keep them as a jest commands, we should: 
- add jest install to prereqs (`npm install -g jest`) 
- drop the `export` since it's not needed -- the setup already handles it

Reply:
Good call! I did not know this was an option 😅  Updated!